projects/com.oracle.truffle.llvm.parser.bc.impl/src/com/oracle/truffle/llvm/parser/bc/impl/LLVMBitcodeVisitor.java

```diff
@@ -353,4 +357,8 @@ public void visit(FunctionDefinition method) {
     @Override
     public void visit(Type type) {
     }
+
+    public LLVMTypeHelper getTypeHelper() {
```

Comment:
I'm not sure what you mean. This method is so that `LLVMBitcodeFunctionVisitor` can get the type helper. Neither of those classes has an `LLVMParserRuntime`.

And the node factories can't be stateless unless you pass in a type helper for each call that needs it.